test/sanity/code-smell/mypy.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+# Copyright: (c) 2020-2021, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""Making sure the type annotations are correct across packages."""
+
+import subprocess
+import sys
+
+
+def run_mypy(python_version=None):
+    """Execute mypy against a given Python version.
+
+    This function proxies mypy's return code and filters out
+    the output so that `ansible-test` wouldn't get confused.
+    """
+    mypy_cmd = (
+        sys.executable, '-m', 'mypy',
+        '--show-error-codes',
+        *(
+            () if python_version is None
+            else ('--python-version', python_version)
+        ),
+        # 'hacking/shippable/incidental.py',
+        'lib/ansible/galaxy/collection/',
+        'lib/ansible/galaxy/dependency_resolution',
+        # 'test/lib/ansible_test/_internal',
+        # 'test/utils/shippable/check_matrix.py',
+    )
+    try:
+        mypy_out = subprocess.check_output(mypy_cmd, universal_newlines=True)
+    except subprocess.CalledProcessError as proc_err:
+        mypy_out = proc_err.output
+        return_code = proc_err.returncode
+    else:
+        return_code = 0
+
+    if not return_code:
+        # NOTE: Interrupt because `ansible-test sanity` runner treats any
+        # NOTE: output as a linting failure.
+        return return_code
+
+    print(
+        'Results of type checking with mypy against '
+        f'{python_version or "unspecified"!s} Python '
```

Comment:
This is allowed because this sanity check module by itself is supposed to be running under Python 3.6.